src/eredis_cluster_monitor.erl

```diff
@@ -359,19 +359,8 @@ parse_cluster_slots([], _Index, Acc) ->
 %% =============================================================================
 -spec get_current_options(State::#state{}) -> options().
 get_current_options(State) ->
-    PasswordEntry = case application:get_env(eredis_cluster, password) of
-                        {ok, Password} -> [{password, Password}];
-                        _ -> []
-                    end,
-    TlsEntry = case application:get_env(eredis_cluster, tls) of
-                   {ok, TlsConf} -> [{tls, TlsConf}];
-                   _ -> []
-               end,
-    SockOptsEntry = case application:get_env(eredis_cluster, socket_options) of
-                        {ok, SockOptsConf} -> [{socket_options, SockOptsConf}];
-                        _ -> []
-                    end,
-    lists:usort(State#state.node_options ++ PasswordEntry ++ TlsEntry ++ SockOptsEntry).
+    Env = application:get_all_env(eredis_cluster),
```

Comment:
Is it possible to change the env. vars during runtime, via set env I guess. One effect by pulling all envs. in might be that at L138 (reload_slots_map) there is a comparison if the node config has changes, and if someone changes a env for something else the node will be reconnected

Reply:
Right... But on the other hand, changes to the pool options will now affects the pools. It can be considered a bugfix.

Do you think we should ignore the pool options in reload_slots_map to avoid reconnections and keep backward compatibility?

(There doesn't seem to be any way to change pool options on a running poolboy pool. That would have been the best option.)

Reply:
I think its good that we have the pool config in the options aswell, since if the option is updated the pool should be closed down since it wont match in L138, and then a new pool is created with the correct size.
I was just thinking about env variables that we dont use in eredis_cluster at all, just something the user sets for something else. If some non-used environment variable is changed or added all pools would be reconnected even that out configs are the same, but maybe it wont be problem. Im not that familiar how the env-variable are propagated from a user, if you thinks ok Im all in. The alternative would be, just as before, get known env./variable but include the the pool options.

Reply:
Ah, those are no problem. `application:get_all_env(eredis_cluster)` only returns our env, not for other applications.